File: payment_study/src/main/java/com/ps/idempot/domain/PaymentRepository.java
package com.ps.idempot.domain;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface PaymentRepository extends JpaRepository<Payment, Long> {

        boolean existsByIdempotency(final String idempotency);

        Optional<Payment> findByIdempotency(final String idempotency);
        }

File: k_rstudy/src/main/kotlin/com/krstudy/kapi/domain/payments/repository/PaymentRepository.kt
package com.krstudy.kapi.domain.payments.repository

import com.krstudy.kapi.domain.payments.entity.Payment
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface PaymentRepository : JpaRepository<Payment, Long> {
    fun findByOrderId(orderId: String): Payment?
}